app/Http/Controllers/ApiPagesController.php

```diff
@@ -41,4 +41,67 @@ public function createNewApp()
     {
         return view('api.pages.mynewapp');
     }
+
+    /**
+     * Generate a token for user.
+     *
+     * @return string
+     */
+    public function generateToken()
+    {
+        $appSecret    = getenv('APP_SECRET');
+        $jwtAlgorithm = getenv('JWT_ALGORITHM');
+        $timeIssued   = time();
+        $serverName   = getenv('SERVERNAME');
+        $tokenId      = base64_encode(getenv('TOKENID'));
+        $token        = [
+            'iss'  => $serverName,   //Issuer: the server name
+            'iat'  => $timeIssued,   // Issued at: time when the token was generated
+            'jti'  => $tokenId,          // Json Token Id: an unique identifier for the token
+            'nbf'  => $timeIssued, //Not before time
+            'exp'  => $timeIssued + 60 * 60 * 24 * 30, // expires in 30 days
+        ];
+        
+        return JWT::encode($token, $appSecret, $jwtAlgorithm);
+    }
+
+    /**
+     * This method post the details from the user into the database
+     *
+     */
+    public function postAppDetails(Request $request)
+    {
+        $this->validate($request, [
+            'name'         => 'required',
+            'homepage_url' => 'required|url',
+            'description'  => 'required',
+        ]);
+
+        AppDetail::create([
+        'name'         => $request->name,
+        'user_id'      => auth()->user()->id,
+        'homepage_url' => $request->homepage_url,
+        'description'  => $request->description,
+        'api_token'    => $this->generateToken(),
+        ]);
+
+        return redirect('/developer/myapp/app-detail');
```

Comment:
**Code Quality Nitpick**
@andela-araimi It is always better and looks neater when you pass a route's name to a redirect method. Check this out https://laravel.com/docs/master/routing#named-routes on how to use named routes in your routes file.